lib/filter.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const urlFn = require('url');
```

Comment:
It's simply for aesthetic, but you can refactor to `const { parse } = require('url');`, then line 7 (& 9) can be just `const data = parse(url);`.